### PR TITLE
General Tool Upgrade

### PR DESCRIPTION
**Pull request by perrito666**
* Tools that are used for scanning were upgraded to latest version available
* Ubuntu version used for all processes, including docker images were upgraded to 22.04, latest LTS without excessive use of snap.
* arm64/amd64 scripts were clearly defined and we tried to avoid repetition where possible.
* Python in use is 3.10 now

This was so far tested only with checkov.
AppImage still needs some work to be fully reliable.

### CODE REVIEW

File: appimage-reqs.sh [hunk 1]
```diff
@@ -1,79 +1,219 @@
...
+
+## Download and install kubesec (https://github.com/controlplaneio/kubesec)
+K8SSEC_TAR="kubesec_linux_${ARCH_ALT_NAME}.tar.gz"
+if [ ! -f "${K8SSEC_TAR}" ]; then
+    echo "Downloading ${K8SSEC_TAR}"
+    curl -LO "https://github.com/controlplaneio/kubesec/releases/download/v${KUBESEC_VERSION}/${K8SSEC_TAR}"
+fi
+    echo "Installing ${K8SSEC_TAR}"
+    tar -C "${USR_BIN_PATH}" -xzvf "${K8SSEC_TAR}"
+    mayberm "${K8SSEC_TAR}"
```
Comment (Ferada) [inline appimage-reqs.sh:165]:
```suggestion
echo "Installing ${K8SSEC_TAR}"
tar -C "${USR_BIN_PATH}" -xzvf "${K8SSEC_TAR}"
mayberm "${K8SSEC_TAR}"
```

File: appimage-reqs.sh [hunk 2]
```diff
@@ -1,79 +1,219 @@
...
-    && mv findsecbugs-plugin-${FSB_VERSION}.jar ${OPTDIR}/spotbugs-${SB_VERSION}/plugin/findsecbugs-plugin.jar \
-    && curl -LO "https://repo1.maven.org/maven2/com/mebigfatguy/sb-contrib/sb-contrib/${SB_CONTRIB_VERSION}/sb-contrib-${SB_CONTRIB_VERSION}.jar" \
-    && mv sb-contrib-${SB_CONTRIB_VERSION}.jar ${OPTDIR}/spotbugs-${SB_VERSION}/plugin/sb-contrib.jar \
-    && mv ${OPTDIR}/spotbugs-${SB_VERSION} ${OPTDIR}/spotbugs \
-    && php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" && php composer-setup.php \
-    && mv composer.phar ${APPDIR}/usr/bin/composer \
-    && rm composer-setup.php
+## Copy the python application code into the AppDir if APPIMAGE is set
+if [ -n "${APPIMAGE}" ]; then
+  cp scan lib tools_config "${APPDIR}"/usr/src -r
```
Comment (Ferada) [inline appimage-reqs.sh:216]:
```suggestion
  cp -r scan lib tools_config "${APPDIR}"/usr/src
```

File: requirements.txt [hunk 3]
```diff
@@ -2,16 +2,16 @@ sarif-om
 jschema_to_python
 defusedxml
 gitpython
-shiftleft-scan-reports
+git+https://github.com/perrito666/scan-reports.git
```
Comment (Ferada) [inline requirements.txt:5]: That's going to be there permanently?

File: ubuntu_build.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# Build the cache folder if missing
+mkdir -p appimage-builder-cache
+
+# Download latest AppImage Builder Tool
+wget https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-"${ARCH}".AppImage
+chmod +x appimagetool-"${ARCH}".AppImage
+./appimagetool-"${ARCH}".AppImage --appimage-extract
+ln -s /squashfs-root/AppRun ./appimage-builder-cache/appimagetool
+
+# Dwonload latest AppImage Builder Runtime
```
Comment (Ferada) [inline ubuntu_build.sh:28]:
```suggestion
# Download latest AppImage Builder Runtime
```

File: requirements.txt (hunk 3 above)
Comment (gacevedo) [reply requirements.txt:5]: `scan-reports` repo wasn't public. It is now, though (http://github.com/ShiftLeftSecurity/scan-reports).

Comment (gacevedo) [reply requirements.txt:5]: We also own the entry in PyPI.

File: .github/workflows/pythonapp.yml [hunk 5]
```diff
@@ -4,11 +4,11 @@ on: [push]
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: [3.8, 3.9]
+        python-version: [3.10.12]
```
[the remark this answers is not in the document]
Comment (perrito666) [reply .github/workflows/pythonapp.yml:11]: No, github interprets i want 3.1 and fails

File: .github/workflows/pythonapp.yml [hunk 6]
```diff
@@ -17,7 +17,7 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
+        python3 -m pip install --upgrade pip
```
[the remark this answers is not in the document]
Comment (perrito666) [reply .github/workflows/pythonapp.yml:20]: not always but there is a number of cases where it is, better to be specific. every place where current python is available, there is a python3 executable

File: Dockerfile [hunk 7]
```diff
@@ -1,96 +1,81 @@
...
-    && rm spotbugs-${SB_VERSION}.tgz \
-    && curl -LO "https://repo1.maven.org/maven2/com/h3xstream/findsecbugs/findsecbugs-plugin/${FSB_VERSION}/findsecbugs-plugin-${FSB_VERSION}.jar" \
-    && mv findsecbugs-plugin-${FSB_VERSION}.jar /opt/spotbugs-${SB_VERSION}/plugin/findsecbugs-plugin.jar \
-    && curl -LO "https://repo1.maven.org/maven2/com/mebigfatguy/sb-contrib/sb-contrib/${SB_CONTRIB_VERSION}/sb-contrib-${SB_CONTRIB_VERSION}.jar" \
-    && mv sb-contrib-${SB_CONTRIB_VERSION}.jar /opt/spotbugs-${SB_VERSION}/plugin/sb-contrib.jar \
-    && curl "https://cdn.shiftleft.io/download/sl" > /usr/local/bin/shiftleft/sl \
-    && chmod a+rx /usr/local/bin/shiftleft/sl
+COPY appimage-reqs.sh /
+COPY requirements.txt /
+COPY shiftleft_scan_reports-1.5.1-py3-none-any.whl /
```
[the remark this answers is not in the document]
Comment (perrito666) [reply Dockerfile:31]: actually I need to remove that line, tx for the reminder

File: appimage-builder-arm64.yml [hunk 8]
```diff
@@ -1,29 +1,8 @@
...
-  - chmod +x AppDir/usr/src/appimage-reqs.sh && AppDir/usr/src/appimage-reqs.sh AppDir
-  - npm install --no-audit --progress=false --only=production --no-save --prefix AppDir/usr/local/lib yarn @cyclonedx/cdxgen @microsoft/rush
-  - ln -s /usr/bin/sh AppDir/usr/bin/sh && ln -s /usr/bin/bash AppDir/usr/bin/bash && ln -s ./AppDir/usr/bin/python3 ./AppDir/usr/bin/python
-  - cp -r /usr/share/icons/Adwaita AppDir/usr/share/icons/
-  - cp AppDir/usr/share/icons/Adwaita/scalable/apps/utilities-terminal-symbolic.svg /build/AppDir/usr/share/icons/utilities-terminal.svg
-  - cp AppDir/usr/share/icons/Adwaita/scalable/apps/utilities-terminal-symbolic.svg AppDir/usr/share/icons/Adwaita/scalable/apps/utilities-terminal.svg 
-  - mkdir -p AppDir/opt/phpsast && cd AppDir/opt/phpsast && composer init --name shiftleft/scan --description scan --quiet && composer require --quiet --no-cache -n --no-ansi --dev vimeo/psalm:^4.1
-  - composer require --quiet --no-cache --dev phpstan/phpstan
-
+  - chmod +x ./appimage-reqs.sh && ./appimage-reqs.sh $(pwd)/AppDir
```
[the remark this answers is not in the document]
Comment (perrito666) [reply appimage-builder-arm64.yml:3]: some of the underlying tools have.... issues, finding the path, this thing does all kinds of odd things with chroots

File: appimage-builder.yml [hunk 9]
```diff
@@ -57,7 +49,7 @@ AppDir:
     env:
       PATH: '${APPDIR}/usr/bin:${APPDIR}/usr/bin/nodejs/bin:${PATH}:${APPDIR}/opt/phpsast/vendor/bin:${APPDIR}/usr/local/lib/node_modules/.bin:'
       PYTHONHOME: '${APPDIR}/usr'
-      PYTHONPATH: '${APPDIR}/usr/lib/python3.8/site-packages:$APPDIR/usr/lib/python3.8:$APPDIR/usr/lib/python3.8/lib-dynload'
+      PYTHONPATH: '${APPDIR}/usr/lib/python3.8/site-packages:$APPDIR/usr/lib/python3.10:$APPDIR/usr/lib/python3.8/lib-dynload'
```
[the remark this answers is not in the document]
Comment (perrito666) [reply appimage-builder.yml:52]: this is a bug

File: requirements.txt (hunk 3 above)
Comment (perrito666) [reply requirements.txt:5]: Nope, did you merge my changes?

File: .github/workflows/pythonapp.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Ferada) [reply .github/workflows/pythonapp.yml:11]: hahaha. ok

**Pull request merged**